azkaban-common/src/main/java/azkaban/server/session/SessionCache.java

```diff
@@ -46,13 +52,16 @@
    * Constructor taking global props.
    */
   @Inject
-  public SessionCache(final Props props) {
+  public SessionCache(final Props props, @Nullable final CommonMetrics metrics) {
     this.effectiveSessionTimeToLive = props.getLong(ConfigurationKeys.SESSION_TIME_TO_LIVE,
         DEFAULT_SESSION_TIME_TO_LIVE);
     this.cache = CacheBuilder.newBuilder()
         .maximumSize(props.getInt("max.num.sessions", MAX_NUM_SESSIONS))
-        .expireAfterAccess(effectiveSessionTimeToLive, TimeUnit.MILLISECONDS)
+        .expireAfterAccess(this.effectiveSessionTimeToLive, TimeUnit.MILLISECONDS)
         .build();
+    if (metrics != null) {
+      metrics.addSessionCount(this::getSessionCount);
```

Comment:
Have you thought about registering this metric after SessionCache's initialization, outside its constructor? Maybe SessionCache doesn't need to know about metrics existence.

Reply:
good point, updated.